jobs/keepalived/templates/bin/action.sh.erb

```diff
@@ -32,5 +37,24 @@ if [[ "$iaas_name" == "aws" ]]; then
     sudo ip addr del $virtual_ip/32 dev eth0
     set -e
   fi
+elif [[ "$iaas_name" == "openstack"  ]]; then
+  # shellcheck disable=SC1091
+  source /var/vcap/jobs/keepalived/bin/utils/openstack/generate_token
+  source /var/vcap/jobs/keepalived/bin/utils/openstack/port_operations
+  this_port_id=$(get_port_id "${SOURCE_IP}")
+  if [[ "$STATE" == "MASTER" ]]; then
+    port_update_response=$(update_port "${this_port_id}")
+    peer_port_id=$(get_port_id "${PEER_IP}")
+    port_update_response=$(update_port "${peer_port_id}")
+    port_update_response=$(update_port "${this_port_id}" "${virtual_ip}")
+    if [[ "$(sudo ip addr show | grep "${virtual_ip}")" != *"${virtual_ip}"* ]]; then
+       sudo ip addr add "${virtual_ip}" dev eth0
+    fi
+  elif [[ "$STATE" == "BACKUP" ]] || [[ "$STATE" == "FAULT" ]]; then
+    port_update_response=$(update_port "${this_port_id}")
+    set +e
```

Comment:
this code is same as in line 36-38. Can we move it to a common function, say, `delete_ip_addr` ?

Reply:
done